api/src/main/java/com/epam/pipeline/manager/pipeline/PipelineRunAsManager.java

```diff
@@ -146,9 +150,45 @@ private PipelineRun runWithUserContext(final String runAsUserName, final Callabl
     private void configureRunnerSids(final PipelineStart runVO, final String runAsUser) {
         final String currentUser = authManager.getAuthorizedUser();
         final RunnerSid allowedRunnerSid = userRunnersManager.findRunnerSid(currentUser, runAsUser);
+        validateRunner(runVO, runAsUser, allowedRunnerSid, currentUser);
+        runVO.setRunSids(buildRunSids(runVO.getRunSids(), currentUser, allowedRunnerSid.getAccessType()));
+    }
+
+    private void validateRunner(final PipelineStart runVO,
+                                final String runAsUser,
+                                final RunnerSid allowedRunnerSid,
+                                final String currentUser) {
         Assert.notNull(allowedRunnerSid, messageHelper.getMessage(
                 MessageConstants.ERROR_RUN_ALLOWED_SID_NOT_FOUND, runAsUser));
-        runVO.setRunSids(buildRunSids(runVO.getRunSids(), currentUser, allowedRunnerSid.getAccessType()));
+        final Long pipelineId = runVO.getPipelineId();
+        if (Objects.nonNull(pipelineId)) {
+            Assert.isTrue(isNullOrTrue(allowedRunnerSid.getPipelinesAllowed()),
+                    messageHelper.getMessage(
+                            MessageConstants.ERROR_RUN_AS_PIPELINES_NOT_ALLOWED, currentUser, runAsUser));
+            if (CollectionUtils.isNotEmpty(allowedRunnerSid.getPipelines())) {
+                Assert.isTrue(allowedRunnerSid.getPipelines().contains(pipelineId),
+                        messageHelper.getMessage(MessageConstants.ERROR_RUN_AS_PIPELINE_NOT_ALLOWED,
+                                currentUser, pipelineId, runAsUser));
+            }
```

Comment:
I would also extract `validatePipeline` method here just to be consistent with `validateTool`.

Reply:
Sure, extracted method